inductiva/coastal/bathymetry.py

```diff
@@ -355,28 +370,40 @@ def plot(
             path: Path to save the plot. If `None`, the plot is not saved, and
               the matplotlib `Axes` object is returned instead.
             x_resolution: Resolution, in meters, of the plotting grid in the x
-              direction.
+              direction. If `None`, the resolution is determined by the
+              bathymetry
             y_resolution: Resolution, in meters, of the plotting grid in the y
-              direction.
+              direction. If `None`, the resolution is determined by the
+              bathymetry
             threshold_distance: Threshold distance to filter out points on the
               uniform grid that are far from points where the bathymetry is
               defined.
         """
 
-        logging.info("Plotting the bathymetry on a uniform grid...")
-        depths_grid, _ = self._interpolate_to_uniform_grid(
-            x_resolution,
-            y_resolution,
-            threshold_distance=threshold_distance,
-            nullable=True,
-        )
+        if self.is_uniform_grid(
+        ) and x_resolution is None and y_resolution is None:
+            x_size = self.x_num()
+            y_size = self.y_num()
 
-        x_size = self.x_ptp() / x_resolution
-        y_size = self.y_ptp() / y_resolution
+            depths_grid = self.depths.reshape((x_size, y_size))
+
+        else:
+            logging.info("Plotting the bathymetry on resized uniform grid...")
+
+            depths_grid, _ = self._interpolate_to_uniform_grid(
+                x_resolution,
+                y_resolution,
+                threshold_distance=threshold_distance,
+                nullable=True,
+            )
+
+            x_size = self.x_ptp() / x_resolution
+            y_size = self.y_ptp() / y_resolution
 
         if x_size > 1000 and y_size > 1000:
```

Comment:
I don't think you need this....

Reply:
You mean the warning for large x and y sizes?

Reply:
yes... why not 800 x 800 or 2000 x 2000?
If we don't have a good number, we don't do it.
If we want to educate the user, then we measure the time a log.info how long it took so that the user then decides...